registry/config.go

```diff
@@ -331,10 +331,6 @@ func (config *ServiceConfig) NewRepositoryInfo(reposName string) (*RepositoryInf
 
 	if repoInfo.Index.Official {
 		normalizedName := repoInfo.RemoteName
-		if strings.HasPrefix(normalizedName, "library/") {
-			// If pull "library/foo", it's stored locally under "foo"
-			normalizedName = strings.SplitN(normalizedName, "/", 2)[1]
```

Comment:
This would mean that if you `docker pull busybox`, you'll have `busybox` locally, but if you then `docker pull library/busybox`, you'll get the same image under `library/busybox` locally.

The original idea behind this change (as I understand it) was that nobody ever sees `library/busybox` locally because that literally is the `busybox` image (compare https://registry.hub.docker.com/_/busybox/ to https://registry.hub.docker.com/u/library/busybox/ -- same exact page because they're the same image).  While normalizing this is a noble goal, it really is a Hub top-level-namespace implementation detail leaking into the Docker source.

Where this is most important is that right now, the server that builds `library/busybox` can't tag as that with 1.6.0 because it gets normalized to just `busybox` automatically, but it has to be `library/busybox` in order to actually be pushed, so we're having to do further namespacing hacks to get that working.